src/twisted/test/test_defer.py

```diff
@@ -2191,77 +2297,88 @@ def testLock(self):
         firstUnique = object()
         secondUnique = object()
 
-        controlDeferred = defer.Deferred()
+        controlDeferred: Deferred[object] = Deferred()
 
-        def helper(self, b):
-            self.b = b
+        result: Optional[object] = None
```

Comment:
Use a variable (`nonlocal` in `helper()`) instead of setting an attribute on `self`.